src/interpreter.c

```diff
@@ -606,8 +609,11 @@ static jl_value_t *eval_body(jl_array_t *stmts, interpreter_state *s, size_t ip,
                     // equivalent to jl_pop_handler(hand_n_leave), longjmping
                     // to the :enter code above instead, which handles cleanup
                     jl_handler_t *eh = ct->eh;
-                    while (--hand_n_leave > 0)
+                    while (--hand_n_leave > 0) {
+                        // pop GC frames for any skipped handlers
```

Comment:
The last frame doesn't include the PUSH'd frame (at `jl_enter_handler(ct, &__eh);` above), so I think you may need to move the `jl_enter_handler(ct, &__eh);` later?

Reply:
This is the second-to-last frame though